client/src/language/utils.ts

```diff
@@ -59,64 +59,3 @@ export const getIndentationOnLine = (document: TextDocument, lineNumber: number)
   const line = document.lineAt(lineNumber)
   return line.text.slice(0, line.firstNonWhitespaceCharacterIndex)
 }
-
-export const checkIsDefinitionUriEqual = (definition: Location | LocationLink, uri: Uri): boolean => {
```

Comment:
I propose this alternate organization:

language/utils/embeddedLanguageDocuments.ts
language/utils/definition.ts
language/utils/textDocument.ts (for the remaining `getIndentationOnLine` function)

Otherwise the utils.ts file is mess of "misc" content with no clearly defined scope.

(I put the "definition functions" in middlewareDefinition after having wasted too much time trying to reuse them for middlewareReferences)